blocks/aem-asset-selector/aem-asset-selector-util.js

```diff
@@ -55,62 +155,212 @@ export function init(cfg, callback) {
   });
 }
 
-function onClose() {
-  document.getElementById('asset-selector-dialog').close();
-}
-
+/**
+ * Generates a URL that can be used to retrieve an asset's binary
+ * without authentication.
+ * @param {string} url URL to the asset in AEM.
+ * @returns {Promise} Resolves with the public asset URL.
+ */
 async function getAssetPublicUrl(url) {
   const response = await fetch(`${WEB_TOOLS}/asset-bin?src=${url}`, {
     headers: {
       Authorization: `Bearer ${imsInstance.getImsToken()}`,
       'x-api-key': API_KEY,
     },
   });
-  if (response && response.ok) {
-    const json = await response.json();
-    return json['asset-url'];
+  if (!response) {
+    throw new Error('Did not receive response from request');
   }
-  if (response && !response.ok) {
-    throw new Error(response.statusTest);
+  if (!response.ok) {
+    throw new Error(`Request failed with status ${response.status}: ${response.statusText}`);
   }
-  return null;
+  const json = await response.json();
+  return json['asset-url'];
 }
 
-async function handleSelection(selection) {
-  const selectedAsset = selection[0];
+/**
+ * Retrieves the rendition that should be copied into the clipboard for
+ * the given asset.
+ * @param {Asset} asset Asset whose copy rendition should be retrieved.
+ * @returns {Rendition} Rendition whose content should be copied to
+ *  the clipboard. Will return a falsy value if no suitable rendition
+ *  could be found.
+ */
+function getCopyRendition(asset) {
   let maxRendition = null;
-  // eslint-disable-next-line no-underscore-dangle
-  selectedAsset._links['http://ns.adobe.com/adobecloud/rel/rendition'].forEach((rendition) => {
-    if ((!maxRendition || maxRendition.width < rendition.width)) {
-      maxRendition = rendition;
-    }
-  });
+  const renditions = getRel(asset, REL_RENDITIONS);
+  if (!renditions) {
+    return maxRendition;
+  }
+  renditions
+    .filter((rendition) => rendition.type === 'image/png')
```

Comment:
Is the `rendition.type === 'image/png'` check necessary? With it, we'd be omitting the better resolution renditions. I recollect having added it since clipboard api did not support other mime types. Do we still have that limitation?

Reply:
The clipboard API only supports PNGs as blobs, so I added that for expedience. I'll add a `TODO` that we should revisit this and create a follow-up tickiet.

Reply:
ok, I think the best png rendition we have is 319px which may become quite blurry on rendering. I thought the signed url from RAPI does not require auth headers but apparently it does need.  One thing worth trying could be [local object urls](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static).

Reply:
Wouldn't Franklin convert the images anyway into webp (or other formats/sizes depending on the js)? SO having PNG at least should be fine to start with an MVP.

Reply:
> Wouldn't Franklin convert the images anyway into webp 
Yes it would but what I am not sure is whether it'd improve the image in anyway if it input image is a 319px png rendition. I agree that we can park it and tackle in a separate issue but overall seems to be an important requirement for real customer use cases.

Reply:
Agreed that this needs to be improved. I put it in there as a "placeholder" without doing a great job communicating that it needs to be improved :)

I've created #48 as a follow-up to this.